apps/src/lab2/levelEditors/aichatSettings/ModelSelectionFields.tsx

```diff
@@ -0,0 +1,117 @@
+import React, {useContext, useState} from 'react';
+
+import {BodyFourText} from '@cdo/apps/componentLibrary/typography';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox/Checkbox';
+import SimpleDropdown from '@cdo/apps/componentLibrary/dropdown/simpleDropdown';
+
+import {modelDescriptions} from '@cdo/apps/aichat/constants';
+import moduleStyles from './edit-aichat-settings.module.scss';
+import {UpdateContext} from './UpdateContext';
+import FieldSection from './FieldSection';
+import {Visibility} from '@cdo/apps/aichat/types';
+
+const unionModelIds = (
+  additionalIds: Set<string>,
+  selectedId: string
+): string[] => {
+  return Array.from(new Set(additionalIds).add(selectedId));
+};
+
+const ModelSelectionFields: React.FunctionComponent = () => {
+  const {setModelSelectionValues, aichatSettings} = useContext(UpdateContext);
+  const shouldDisableAdditionalModelSelection =
+    aichatSettings.visibilities.selectedModelId !== Visibility.EDITABLE;
+  const [additionalAvailableModelIds, setAdditionalAvailableModelIds] =
+    useState<Set<string>>(
+      new Set(
+        aichatSettings.availableModelIds.filter(
+          id => id !== aichatSettings.initialCustomizations.selectedModelId
+        )
+      )
+    );
+  const selectedModelId = aichatSettings.initialCustomizations.selectedModelId;
+
+  return (
+    <FieldSection
+      fieldName="selectedModelId"
+      labelText="Model Selection"
+      description="The initial model selected for the chatbot."
+      inputType="custom"
+      inputNode={
+        <>
+          <SimpleDropdown
+            labelText=""
+            onChange={e => {
+              setModelSelectionValues(
+                unionModelIds(additionalAvailableModelIds, e.target.value),
```

Comment:
Do we need to union these each time before calling `setModelSelectionValues`? Or could we just pass in the available and selected to `setModelSelectionValues`, union them inside that function, then set state appropriately?

Reply:
Oh good call, that would be more DRY.